sw/host/tests/rom/e2e_bootup_bad_rom_ext_signature/src/main.rs

```diff
@@ -53,12 +59,12 @@ fn test_corrupted_rom_ext(opts: &Opts, transport: &TransportWrapper) -> Result<(
 
     // TODO: Should really `opts.init.uart_params.create()` here, but we need to refactor
     // BootstrapOptions first.
-    //let uart = opts.init.uart_params.create(&transport)?;
     let uart = transport.uart("0")?;
-    // This is a negative test, so a boot failure is expected
+    // This is a negative test, so a boot failure with error code   kErrorSigverifyBadEncodedMessage`
+    // is expected.
     let mut console = UartConsole {
         timeout: Some(opts.timeout),
-        exit_success: Some(Regex::new(r"BFV:[0-9a-z]{8}\r\nLCV:[0-9a-z]{8}\r\n")?),
+        exit_success: Some(Regex::new(r"BFV:01535603\r\nLCV:2739ce73\r\n")?),
```

Comment:
Since I'm not familiar with the project yet, could you tell me where these magic numbers come from?

Reply:
See https://github.com/lowRISC/opentitan/blob/master/sw/device/silicon_creator/lib/error.h and lc_ctrl_regs.h

Reply:
These numbers are error codes defined in `opentitan/sw/device/silicon_creator/lib/error.h`. As this is a test that checks that the rom is returning the correct error code, and we want the test to fail wherever theses numbers change I hardcoded it.

Reply:
Why is still this using a regex when it's now a simple (sub?)string comparison?
Ideally we'd hardcode the error components from which the error number is generated instead of the error number itself, but I assume we don't have a Rust implementation of `error.h`?

Reply:
> Why is still this using a regex when it's now a simple (sub?)string comparison? 

This has to be a regex because the type of `exit_success` is `Option<Regex>`.

Ideally we'd hardcode the error components from which the error number is generated instead of the error number itself, but I assume we don't have a Rust implementation of `error.h`?

Yes, we don't and the way error codes are generated are opaque to the test harness. Please note that matching the BFV and LCV values are just one use case. We also have tests that match other patterns printed by the test programs that run from flash.